### PR TITLE
fix(repository): fixed entitytoremove=null error with mongodb. 

### LINKED ISSUE #43: findById return EntityNotFound when we using mongodb
When we using loopback4 with database mongodb
We also using id generate by MongoDb with field name _id, without using id field
In SoftCrudRepository into this package , when findById, it using id only

`//As parent method findById have filter: FilterExcludingWhere<T>
    //so we need add check here.
    const entityToRemove = await super.findOne(filter, options);

    if (entityToRemove) {
      // Now call super
      return super.findById(id, filter, options);
    } else {
      throw new HttpErrors.NotFound(ErrorKeys.EntityNotFound);
    }`
That take entityToRemove always null then it return an error

### PR DESCRIPTION
**Pull request by faizan1620**
## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #43 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Intermediate change (work in progress)

## How Has This Been Tested ?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [x] Performed a self-review of my own code
- [ ] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the style guide
- [ ] API Documentation in code was updated

### CODE REVIEW

**Comment by faizan1620** [comment]
fixed entitytoremove=null error with mongodb.

**Review by yeshamavani — APPROVED** [review]

File: src/repositories/soft-crud.repository.base.ts [hunk 1]
```diff
@@ -119,11 +119,25 @@ export abstract class SoftCrudRepository<
     return super.findOne(filter, options);
   }
 
+  //To find the primary key/Unique Id field name
+  async getPkFieldName() {
+    var data = this.entityClass.definition.properties;
+    for (let key in data) {
```
Comment (akshatdubeysf) [inline src/repositories/soft-crud.repository.base.ts:125]: Why is this an async function?

File: src/repositories/soft-crud.repository.base.ts [hunk 2]
```diff
@@ -154,10 +166,12 @@ export abstract class SoftCrudRepository<
       filter = filter ?? {};
       filter.where = {
         deleted: false,
-        id: id,
       } as Condition<T>;
     }
 
+    //adding primary key property in where filter
+    filter.where[pk] = id;
```
Comment (akshatdubeysf) [inline src/repositories/soft-crud.repository.base.ts:173]: For cases where and/or operators are used?

**Review by akshatdubeysf — CHANGES_REQUESTED** [review]

**Review by samarpan-b — CHANGES_REQUESTED** [review]
Can we do it similar to how LB4 does it in repository class ? They take it in generic params.

File: src/repositories/soft-crud.repository.base.ts (hunk 1 above)
Comment (faizan1620) [reply src/repositories/soft-crud.repository.base.ts]: done

File: src/repositories/soft-crud.repository.base.ts (hunk 2 above)
Comment (faizan1620) [reply src/repositories/soft-crud.repository.base.ts]: done

**Comment by akshatdubeysf** [comment]
@samarpan-b we have done that, but that gives us the type of the ID, not the name of the ID field.

**Review by akshatdubeysf — APPROVED** [review]

**Pull request merged**